crates/evm/core/src/utils.rs

```diff
@@ -113,37 +110,83 @@ pub fn configure_tx_env(env: &mut revm::primitives::Env, tx: &Transaction<AnyTxE
-
-        if let Some(eip712_meta) = eip712_tx.eip712_meta {
-            metadata = foundry_zksync_core::ZkTransactionMetadata {
-                factory_deps: eip712_meta.factory_deps,
-                paymaster_data: eip712_meta.paymaster_params,
-            };
-        }
+
+    // Extract fields from the raw zkSync transaction format
+    let common_data = &outer_tx["common_data"]["L2"];
+    let execute = &outer_tx["execute"];
+
+    // Set basic transaction fields
+    env.tx.caller = common_data["initiatorAddress"].as_str().unwrap().parse().unwrap();
+    env.tx.transact_to =
+        TxKind::Call(execute["contractAddress"].as_str().unwrap().parse().unwrap());
+    env.tx.gas_limit = U256::from_str_radix(
+        common_data["fee"]["gas_limit"].as_str().unwrap().trim_start_matches("0x"),
+        16,
+    )
+    .unwrap()
+    .try_into()
+    .unwrap();
+    env.tx.nonce = Some(common_data["nonce"].as_u64().unwrap());
+    env.tx.value =
+        U256::from_str_radix(execute["value"].as_str().unwrap().trim_start_matches("0x"), 16)
+            .unwrap();
+    env.tx.data = alloy_primitives::hex::decode(
+        execute["calldata"].as_str().unwrap().trim_start_matches("0x"),
+    )
+    .unwrap()
+    .into();
+    env.tx.gas_price = U256::from_str_radix(
+        common_data["fee"]["max_fee_per_gas"].as_str().unwrap().trim_start_matches("0x"),
+        16,
+    )
+    .unwrap();
+    env.tx.gas_priority_fee = Some(
+        U256::from_str_radix(
+            common_data["fee"]["max_priority_fee_per_gas"]
+                .as_str()
+                .unwrap()
+                .trim_start_matches("0x"),
+            16,
+        )
+        .unwrap(),
+    );
+
+    // Set zkSync specific metadata
+    if let Some(paymaster_params) = outer_tx.get("paymasterParams") {
+        metadata = foundry_zksync_core::ZkTransactionMetadata {
+            factory_deps: execute["factoryDeps"]
+                .as_array()
+                .unwrap()
+                .iter()
+                .map(|dep| {
+                    dep.as_array().unwrap().iter().map(|b| b.as_u64().unwrap() as u8).collect()
+                })
+                .collect(),
+            paymaster_data: Some(foundry_zksync_core::PaymasterParams {
+                paymaster: paymaster_params["paymaster"].as_str().unwrap().parse().unwrap(),
+                paymaster_input: paymaster_params["paymasterInput"]
+                    .as_array()
+                    .unwrap()
+                    .iter()
+                    .map(|b| b.as_u64().unwrap() as u8)
+                    .collect(),
+            }),
+        };
+    } else {
+        metadata = foundry_zksync_core::ZkTransactionMetadata {
+            factory_deps: execute["factoryDeps"]
+                .as_array()
+                .unwrap()
+                .iter()
+                .map(|dep| {
+                    dep.as_array().unwrap().iter().map(|b| b.as_u64().unwrap() as u8).collect()
```

Comment:
Might be worth reducing or removing `unwrap`s for `expect`s in case this functionality breaks again. Also does this work for both anvil-zksync and sepolia now?. Might wanna include a `unit test` with raw transactions.